File: src/Verticie.cpp
#include "Verticie.hpp"
#include <iostream>
#include <vector>





Verticie::Verticie(){
    this -> id = -1;
    this -> arcos = std::vector<Arco>();
}

Verticie::Verticie(int id, int grauEntrada, int grauSaida){
    this -> id = id;
    this -> grauEntrada = grauEntrada;
    this -> grauSaida = grauSaida;
    this -> arcos = std::vector<Arco>();
}

Verticie::Verticie(int id, std::vector<Arco> arcos){
    this -> id = id;
    this -> arcos = arcos;
}

void Verticie::addArco(Arco arco){
    this -> arcos.push_back(arco);
}

int Verticie::getId(){
    return this -> id;
}

std::vector<Arco> Verticie::getArcos(){
    return this -> arcos;
}

bool Verticie::igual(Verticie b){
    return this -> getId() == b.getId();
}

std::vector<Arco>* Verticie::getArcosRef(){
    return &(this -> arcos);
}

File: src/Verticie.cs


using System.Collections;
using System.Numerics;
using System.Security.AccessControl;
using arco;

namespace verticie;



record Verticie(int id, int grauEntrada, int grauSaida) : IComparable<Verticie> {
    public int Id { get; init; } = id;
    public int GrauEntrada {get; init; } = grauEntrada;
    public int GrauSaida {get; init; } = grauSaida;
    public List<Arco> Arcos {get; init; } = new();
    public void addArco(Arco arco){
        this.Arcos.Add(arco);
    }

    // how to add compare so that the sort function can be used
    public int CompareTo(Verticie? v) {
        if (v is null) {
            return 1; 
        }
        return this.Id - v.Id;
    }

}